llvm/include/llvm/IR/PatternMatch.h

```diff
@@ -2596,6 +2602,13 @@ m_Not(const ValTy &V) {
   return m_c_Xor(m_AllOnes(), V);
 }
 
+template <typename ValTy>
+inline BinaryOp_match<cst_pred_ty<is_all_ones, false>, ValTy, Instruction::Xor,
+                      true>
+m_NotForbidPoison(const ValTy &V) {
+  return m_c_Xor(m_AllOnesForbidPoison(), V);
+}
```

Comment:
missing unit tests?

Reply:
I prefer not to have unit tests for anything that has lit test coverage.

Reply:
(Except for exhaustive unit tests.)

Reply:
Err I was confusing, not the `m_NotForbidPoison`, but for the `AllowPoison` template arg in the `cstval_pred_ty` unittests.

Either way though.